sklearn/model_selection/tests/test_validation.py

```diff
@@ -224,15 +224,18 @@ def score(self, X=None, Y=None):
     def get_params(self, deep=False):
         return {'a': self.a, 'allow_nd': self.allow_nd}
 
+    def _get_tags(self):
+        return {'estimator_type': 'classifier'}
+
 
 # XXX: use 2D array, since 1D X is being detected as a single sample in
 # check_consistent_length
-X = np.ones((10, 2))
+X = np.ones((15, 2))
 X_sparse = coo_matrix(X)
-y = np.array([0, 0, 1, 1, 2, 2, 3, 3, 4, 4])
-# The number of samples per class needs to be > n_splits,
-# for StratifiedKFold(n_splits=3)
-y2 = np.array([1, 1, 1, 2, 2, 2, 3, 3, 3, 3])
+y = np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4])
+# The number of samples per class need to be greater than or equal
+# to the number of splits for stratified cross-validation routines
+y2 = np.array([1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3])
```

Comment:
Is this diff related to this PR?

Reply:
It is (in)directly related with this PR.

Now, `MockClassifier` is properly interpreted as a classifier due to the inclusion of the `_get_tags` method (previously, the `_estimator_type` attribute was not set). Then, by default, the cross-validation routines applied in `sklearn.model_selection.GridSearchCV` and `sklearn.model_selection.RandomizedSearchCV` are stratified. Thus, the number of samples of each class should be greater than the number of splits, and this issue was not considered in some of the tests.

Reply:
Opening another Pr to make that change would most likely make this PR easier for others to review.